cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/PeriodicReplicationService.java

```diff
@@ -417,8 +444,51 @@ private int getIntervalInSeconds() {
         }
     }
 
-    private String constructKey(String suffix) {
-        return getClass().getName() + suffix;
+    String constructKey(String suffix) {
+        return constructKey(getClass(), suffix);
+    }
+
+    static String constructKey(Class<? extends PeriodicReplicationService> prsClass,
+                                       String suffix) {
+        return prsClass.getName() + suffix;
+    }
+
+    @Override
+    public void allReplicationsCompleted() {
+        super.allReplicationsCompleted();
+        setReplicationsPending(this, getClass(), false);
+    }
+
+    /**
+     * Sets whether there are replications pending. This may be because replications are
+     * currently in progress and have not yet completed, or because a previous scheduled
+     * replication didn't take place because the conditions for replication were not met.
+     * @param context
+     * @param prsClass
+     * @param pending true if there is a replication pending, or false otherwise.
+     */
+    public static void setReplicationsPending(Context context, Class<? extends
```

Comment:
It's my understanding that the `replicationsPending` state as stored in shared preferences will persist across reboots. What's the impact of restarting the device when there are still replications pending, is it a problem or will they automatically resume and eventually set `replicationsPending` to `false`?